subprojects/gradle/buildchecks/src/test/kotlin/com/avito/android/plugin/build_param_check/incremental_check/IncrementalKaptTaskTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.avito.android.plugin.build_param_check.incremental_check
+
+import com.avito.test.gradle.AndroidAppModule
+import com.avito.test.gradle.TestProjectGenerator
+import com.avito.test.gradle.gradlew
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+internal class IncrementalKaptTaskTest {
```

Comment:
We have to check:
- without room and with unsupported java

Reply:
There are already at least 3 tests with unsupported Java and one test without Room.

Reply:
I see, but there is not a test I talked about. Maybe you have an opinion why we don't need that test?

Reply:
I didn't get you. Look:
- `build success - room not applied` this test checks that everything is fine when room is not applied.
- `build success with no warnings - unsupported Java version`, `build success with warning - unsupported Java version` and `build fail - unsupported Java version` check behavior on unsupported Java versions.

What tests do you want to see here?

Reply:
`build success - without room and unsupported java`

Reply:
As I understand in your scenarios:
build success with no warnings - unsupported Java version
build success with a warning - unsupported Java version
build fail - unsupported Java version

Room enabled by default

Reply:
Ok. Now I understand. I add a test for disabled Room and unsupported Java. It should leed to a successful build.

Reply:
Add test